src/NServiceBus.Metrics.ServiceControl/QueueLength/IQueueLengthProbe.cs

```diff
@@ -0,0 +1,15 @@
+﻿namespace NServiceBus.Metrics.ServiceControl
+{
+    /// <summary>
+    /// A metrics probe for queue length
+    /// </summary>
+    public interface IQueueLengthProbe
```

Comment:
What if certain queues fail to probe?

Reply:
Also, the current probes are push driven, as they push a value for every message received. That doesn't make sense for queue length in that same sense. Yes, it could push a value every fixed interval but is that a logical model?

Reply:
Well, I see that currently the API is, return -1 if there is a failure.